### ISSUE TITLE
Proof of Authority  Consensus Algorithm 

### ISSUE THREAD

**Issue by masonforest** [exthereum/rfc#2]
### Title

      OIP: 1002
      Title: Proof of Authority  Consensus Algorithm 
      Author: Mason Fischer <mason@kissr.co>
      Status: Draft
      Type: Standard
      Layer: Consensus
      Created: 2018-04-04

### Abstract

We need to implement Proof of Authority consensus in Exthereum so Exthereum can run in the PoA network.

### Specification
Technical Specification: https://github.com/paritytech/wiki/blob/master/Aura.md
Reference implementation: https://github.com/paritytech/parity/tree/master/ethcore/src/engines/authority_round

**Comment by hayesgm** [comment]
Question: do we currently implement any consensus mechanisms? Do we know where this would fit in to the current code-base?

**Comment by masonforest** [comment]
> do we currently implement any consensus mechanisms?

Not that I know of so yeah, we'd have to figure out where where this fits in to the exiting codebase.

**Comment by hayesgm** [comment]
Yeah, it's likely near the boarder of ExWire and Blockchain. Blockchain was meant to (and generally still is) the piece that takes blocks and collects them together into a chain. ExWire is the piece which asks for new blocks from the network. Blockchain (I think) is the right place to decide what is the best block based on some rules.
